pkg/kubelet/kubelet_network.go

```diff
@@ -121,7 +121,7 @@ func (kl *Kubelet) providerRequiresNetworkingConfiguration() bool {
 	// is used or whether we are using overlay networking. We should return
 	// true for cloud providers if they implement Routes() interface and
 	// we are not using overlay networking.
-	if kl.cloud == nil || kl.cloud.ProviderName() != "gce" || kl.flannelExperimentalOverlay {
+	if kl.cloud == nil || kl.cloud.ProviderName() != "gce" {
```

Comment:
I brought this up in https://github.com/kubernetes/kubernetes/pull/33573#issuecomment-251719452.  It's a problem for OpenShift when run on GCE too, as we use an overlay and don't need Routes created for pod<->pod communication, but the NetworkUnavailable condition is set anyway.

But I think this should be solved differently and more cleanly than repurposing the flannelExperimentalOverlay flag.